### PR TITLE
test: add unit tests for channel upgrade event emission

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #3497 

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/keeper/msg_server.go [hunk 1]
```diff
@@ -1084,12 +1084,25 @@ func (k Keeper) ChannelUpgradeCancel(goCtx context.Context, msg *channeltypes.Ms
...
 	cbs.OnChanUpgradeRestore(ctx, msg.PortId, msg.ChannelId)
 
 	ctx.Logger().Info("channel upgrade cancel succeeded", "port-id", msg.PortId, "channel-id", msg.ChannelId)
 
+	// get channel here again to get latest state after write
+	channel, found = k.ChannelKeeper.GetChannel(ctx, msg.PortId, msg.ChannelId)
+	if !found {
+		return nil, errorsmod.Wrapf(channeltypes.ErrChannelNotFound, "port ID (%s) channel ID (%s)", msg.PortId, msg.ChannelId)
+	}
+	keeper.EmitChannelUpgradeCancelEvent(ctx, msg.PortId, msg.ChannelId, channel, upgrade)
```
Comment (crodriguezvega) [inline modules/core/keeper/msg_server.go:1104]: Nice that you moved this here. I guess we can then [remove it from `WriteUpgradeCancelChannel`](https://github.com/cosmos/ibc-go/blob/bb42c9a14f274c903773211fa072365b3d0ad465/modules/core/04-channel/keeper/upgrade.go#L683C2-L683C31)!

**Review by crodriguezvega — APPROVED** [review]
Excellent work, @charleenfei!

File: modules/core/keeper/msg_server.go (hunk 1 above)
Comment (charleenfei) [reply modules/core/keeper/msg_server.go:1104]: aha, yes, i think that was either missed or maybe the merge left the artifact as i do remember there was a PR to remove all of those at some point... i'll take it out now!

File: modules/core/keeper/msg_server_test.go [hunk 2]
```diff
@@ -919,9 +922,32 @@ func (suite *KeeperTestSuite) TestChannelUpgradeInit() {
...
+						channeltypes.AttributeKeyUpgradeConnectionHops: upgrade.Fields.ConnectionHops[0],
+						channeltypes.AttributeKeyUpgradeVersion:        upgrade.Fields.Version,
+						channeltypes.AttributeKeyUpgradeOrdering:       upgrade.Fields.Ordering.String(),
+						channeltypes.AttributeKeyUpgradeSequence:       fmt.Sprintf("%d", channel.UpgradeSequence),
+					},
+					sdk.EventTypeMessage: {
+						sdk.AttributeKeyModule: channeltypes.AttributeValueCategory,
+					},
+				}
+				ibctesting.AssertEventsLegacy(&suite.Suite, expEvents, events)
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/core/keeper/msg_server_test.go:972]: Any ideas on the difference between the two so I don't have to check the code? 

It seems also odd to have to pass the `Suite` as an argument. An instance of `testing.T` could be passed, or an error return should be fine too. 

In any case, I think these event testing functions has been around for a while. We can give them some attention in a follow up as you suggested! :)

File: modules/core/04-channel/keeper/upgrade.go [hunk 3]
```diff
@@ -669,7 +669,7 @@ func (k Keeper) WriteUpgradeCancelChannel(ctx sdk.Context, portID, channelID str
 		panic(fmt.Errorf("could not find existing channel when updating channel state, channelID: %s, portID: %s", channelID, portID))
 	}
 
-	upgrade, found := k.GetUpgrade(ctx, portID, channelID)
+	_, found = k.GetUpgrade(ctx, portID, channelID)
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/core/04-channel/keeper/upgrade.go]: https://github.com/cosmos/ibc-go/pull/5475/commits/a9203681be259d44d085bae336a4869f67baba7a

File: modules/core/keeper/msg_server_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (charleenfei) [reply modules/core/keeper/msg_server_test.go:972]: the `EventsLegacy` function handles sdk.Events, whereas `Events` handles ABCI events (typed array), we can double check these later agreed

[the remark this answers is not in the document]
Comment (charleenfei) [reply modules/core/keeper/msg_server_test.go:972]: I'll add a follow up PR for error receipt events @DimitrisJim to make reviewing easier! thanks <3

[the remark this answers is not in the document]
Comment (crodriguezvega) [reply modules/core/keeper/msg_server_test.go:972]: `AssertEvents` was added when we were upgrading to SDK v0.50 to be able to assert the events in [this test](https://github.com/cosmos/ibc-go/blob/dc99ee43c1ade80094d56acd00ca16097b308d44/modules/apps/29-fee/keeper/events_test.go#L170), if I remember correctly. Now that we are in SDK v0.50 I think we move to `AssertEvents` and remove `AssertEventsLegacy` and also the `EventsMap` type alias.

[the remark this answers is not in the document]
Comment (DimitrisJim) [reply modules/core/keeper/msg_server_test.go:972]: thanks for context @crodriguezvega! I'll open an issue to keep track of this.

**Pull request merged**